MapboxNavigation/TopBannerViewController.swift

```diff
@@ -0,0 +1,450 @@
+        let leading = informationStackView.leadingAnchor.constraint(equalTo: view.safeLeadingAnchor)
+        let trailing = informationStackView.trailingAnchor.constraint(equalTo: view.safeTrailingAnchor)
+        let bottom = informationStackBottomPin
+        //bottom is taken care of as part of steps TVC show/hide
+        
+        NSLayoutConstraint.activate([top, leading, trailing, bottom])
+    }
+    
+    private func setupInformationStackView() {
+        addInstructionsBanner()
+        informationStackView.addArrangedSubviews(secondaryChildren)
+        for child in informationChildren {
+            child.leadingAnchor.constraint(equalTo: informationStackView.leadingAnchor).isActive = true
+            child.trailingAnchor.constraint(equalTo: informationStackView.trailingAnchor).isActive = true
+        }
+    }
+
+
+    public func displayStepsTable() {
+        dismissStepsTable()
+        
+        guard let progress = routeProgress, let parent = parent else {
+            return
+        }
+        
+        
+        let controller = StepsViewController(routeProgress: progress)
+        controller.delegate = self
+        
+        delegate?.topBanner?(self, willDisplayStepsController: controller)
+        embed(controller, in: stepsContainer) { (parent, child) -> [NSLayoutConstraint] in
+            child.view.translatesAutoresizingMaskIntoConstraints = false
+            let pinningConstraints = child.view.constraintsForPinning(to: self.stepsContainer)
+            let hideConstraints = self.stepsContainerHide
+            
+            return pinningConstraints + hideConstraints + self.stepsContainerConstraints
+        }
+        stepsViewController = controller
+        isDisplayingSteps = true
+        
+        parent.view.layoutIfNeeded()
+        
+        
+        let stepsInAnimation = {
+            NSLayoutConstraint.deactivate(self.stepsContainerHide)
+            NSLayoutConstraint.activate(self.stepsContainerShow)
+            
+            
+            let finally: (Bool) -> Void = { _ in
+                self.delegate?.topBanner?(self, didDisplayStepsController: controller)
+            }
+            
+            UIView.animate(withDuration: 0.35, delay: 0.0, options: [.curveEaseOut], animations: parent.view.layoutIfNeeded, completion: finally)
+        }
+
+        hideSecondaryChildren(completion: stepsInAnimation)
+        
+    }
+    
+    public func dismissStepsTable(completion: CompletionHandler? = nil) {
+        guard let parent = parent, let steps = stepsViewController  else { return }
+        parent.view.layoutIfNeeded()
+        
+        delegate?.topBanner?(self, willDismissStepsController: steps)
+
+
+        NSLayoutConstraint.deactivate(stepsContainerShow)
+        NSLayoutConstraint.activate(stepsContainerHide)
+        
+        let complete = { [weak self] in
+            guard let self = self else {
```

Comment:
```swift
guard let `self` = self
```
was an unwanted compiler bug in Swift <4.2
but apparently,
```swift
guard let self = self
``` 
was a proposal that was introduced in Swift 4.2 so that should work. Cleaner than renaming to strongSelf  👍